src/main/java/com/zendesk/maxwell/util/AbstractConfig.java

```diff
@@ -95,6 +100,29 @@ protected Properties readPropertiesFile(String filename, Boolean abortOnMissing)
 		return p;
 	}
 
+	protected Properties readPropertiesEnv(String envConfig) {
+		LOGGER.debug("Attempting to read env_config param: {}", envConfig);
+		String envConfigJsonRaw = System.getenv(envConfig);
+		if (envConfigJsonRaw != null && envConfigJsonRaw.startsWith("{")) {
+			LOGGER.debug("Parsing envConfig");
+			ObjectMapper mapper = new ObjectMapper();
+			try {
+				Map<String, Object> stringMap = mapper.readValue(envConfigJsonRaw, Map.class);
+				Properties properties = new Properties();
+				stringMap.entrySet().stream()
+						.map(entry -> new AbstractMap.SimpleEntry<>(entry.getKey(), entry.getValue().toString()))
+						.forEach(entry -> properties.put(entry.getKey(), entry.getValue()));
```

Comment:
i can never make heads or tails of this kinda code.  I'm going to trust you that it works.

Reply:
will change to the following:
```
	Properties properties = new Properties();
	for (Map.Entry<String, Object> entry : stringMap.entrySet()) {
		properties.put(entry.getKey(), entry.getValue().toString());
	}
	return properties;
```

The point is to allow it to be an Object to be forgiving in the case where a user specifies a Number instead of a String and turn it into a String from there, the same way a properties file would read it.